web/cobrands/angus/position_map.js

```diff
@@ -0,0 +1,345 @@
+            minResolution: min_resolution,
+            styleMap: get_streetlight_stylemap()
+        });
+        fixmystreet.streetlight_layer = streetlight_layer;
+
+        // A non-interactive layer to display existing street light faults
+        var fault_protocol = new OpenLayers.Protocol.WFS({
+            version: "1.1.0",
+            url:  wfs_url,
+            featureType: wfs_fault_feature,
+            geometryName: "g"
+        });
+        streetlight_fault_layer = new OpenLayers.Layer.Vector("WFS", {
+            strategies: [new OpenLayers.Strategy.BBOX()],
+            protocol: fault_protocol,
+            visibility: false,
+            maxResolution: max_resolution,
+            minResolution: min_resolution,
+            styleMap: get_fault_stylemap()
+        });
+
+        // Set up handlers for selecting/unselecting markers and panning/zooming the map
+        select_feature_control = new OpenLayers.Control.SelectFeature( streetlight_layer );
+        streetlight_layer.events.register( 'featureselected', streetlight_layer, streetlight_selected);
+        streetlight_layer.events.register( 'featureunselected', streetlight_layer, streetlight_unselected);
+        streetlight_layer.events.register( 'loadend', streetlight_layer, layer_loadend);
+        streetlight_layer.events.register( 'visibilitychanged', streetlight_layer, layer_visibilitychanged);
+        fixmystreet.map.events.register( 'zoomend', streetlight_layer, check_zoom_message_visiblity);
+        // Set up handlers for simply hovering over a street light marker
+        hover_feature_control = new OpenLayers.Control.SelectFeature(
+            streetlight_layer,
+            {
+                hover: true,
+                highlightOnly: true,
+                renderIntent: 'temporary'
+            }
+        );
+        hover_feature_control.events.register('beforefeaturehighlighted', null, function(e) {
+            // Don't let marker go from selected->hover state,
+            // as it causes some mad flickering effect.
+            if (e.feature.renderIntent == 'select') {
+                return false;
+            }
+        });
+
+        fixmystreet.map.addLayer(streetlight_layer);
+        fixmystreet.map.addLayer(streetlight_fault_layer);
+        fixmystreet.map.addControl( hover_feature_control );
+        hover_feature_control.activate();
+        fixmystreet.map.addControl( select_feature_control );
+        select_feature_control.activate();
+
+        // Make sure the fault markers always appear beneath the street lights
+        streetlight_fault_layer.setZIndex(streetlight_layer.getZIndex()-1);
+
+        // Show/hide the streetlight layer when the category is chosen
+        $("#problem_form").on("change.category", "select#form_category", function(){
+            var category = $(this).val();
+            if (category == streetlight_category) {
+                streetlight_layer.setVisibility(true);
```

Comment:
There's a race condition here, in that the visibility change causes the "Zoom in" message to be shown underneath the category box, then the server loads the column ID box which appears inbetween those two things, and the map zooms in automatically and the message changes anyway. It'd be good if the message didn't appear until after the zoom/column ID box appearing.